### ISSUE TITLE
Failed: Http2ServerSpec, streams testkit on message timeout

### ISSUE THREAD

**Issue by seglo** [akka/akka-http#3659]
https://jenkins.akka.io:8498/job/akka-http-nightly/AKKA_VERSION=master,Node=akka-http-nightly-node,SCALA_VERSION=2.12.12,jdk=AdoptOpenJDK 11/337/testReport/akka.http.impl.engine.http2/Http2ServerSpec/The_Http_2_server_implementation_should_support_stream_for_request_entity_data_should_fail_stream_if_request_entity_is_not_fully_pulled_when_connection_dies/

```
sbt.ForkMain$ForkError: java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsgClass waiting for class akka.stream.testkit.TestSubscriber$OnError
	at scala.Predef$.assert(Predef.scala:223)
	at akka.testkit.TestKitBase.expectMsgClass_internal(TestKit.scala:569)
	at akka.testkit.TestKitBase.expectMsgType(TestKit.scala:541)
	at akka.testkit.TestKitBase.expectMsgType$(TestKit.scala:540)
	at akka.testkit.TestKit.expectMsgType(TestKit.scala:969)
	at akka.stream.testkit.TestSubscriber$ManualProbe.expectError(StreamTestKit.scala:467)
	at akka.http.impl.engine.ws.ByteStringSinkProbe$$anon$1.expectError(ByteStringSinkProbe.scala:102)
	at akka.http.impl.engine.http2.Http2ServerSpec$$anon$22.<init>(Http2ServerSpec.scala:449)
	at akka.http.impl.engine.http2.Http2ServerSpec.$anonfun$new$30(Http2ServerSpec.scala:435)
	at akka.http.impl.engine.http2.Http2ServerSpec$InWithStoppedStages.$anonfun$inAssertAllStagesStopped$2(Http2ServerSpec.scala:1283)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at akka.stream.testkit.scaladsl.StreamTestKit$.assertAllStagesStopped(StreamTestKit.scala:32)
	at akka.http.impl.engine.http2.Http2ServerSpec$InWithStoppedStages.$anonfun$inAssertAllStagesStopped$1(Http2ServerSpec.scala:1282)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
	at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1076)
	at akka.http.impl.util.WithLogCapturing.$anonfun$withFixture$1(WithLogCapturing.scala:38)
	at akka.testkit.EventFilter.intercept(TestEventListener.scala:119)
	at akka.http.impl.util.WithLogCapturing.withFixture(WithLogCapturing.scala:36)
	at akka.http.impl.util.WithLogCapturing.withFixture$(WithLogCapturing.scala:23)
	at akka.http.impl.util.AkkaSpecWithMaterializer.withFixture(AkkaSpecWithMaterializer.scala:14)
	at org.scalatest.wordspec.AnyWordSpecLike.invokeWithFixture$1(AnyWordSpecLike.scala:1074)
	at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTest$1(AnyWordSpecLike.scala:1086)
	at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
	at org.scalatest.wordspec.AnyWordSpecLike.runTest(AnyWordSpecLike.scala:1086)
	at org.scalatest.wordspec.AnyWordSpecLike.runTest$(AnyWordSpecLike.scala:1068)
	at akka.testkit.AkkaSpec.runTest(AkkaSpec.scala:59)
	at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTests$1(AnyWordSpecLike.scala:1145)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:390)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:427)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:390)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:427)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
	at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
	at org.scalatest.wordspec.AnyWordSpecLike.runTests(AnyWordSpecLike.scala:1145)
	at org.scalatest.wordspec.AnyWordSpecLike.runTests$(AnyWordSpecLike.scala:1144)
	at akka.testkit.AkkaSpec.runTests(AkkaSpec.scala:59)
	at org.scalatest.Suite.run(Suite.scala:1112)
	at org.scalatest.Suite.run$(Suite.scala:1094)
	at akka.testkit.AkkaSpec.org$scalatest$wordspec$AnyWordSpecLike$$super$run(AkkaSpec.scala:59)
	at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$run$1(AnyWordSpecLike.scala:1190)
	at org.scalatest.SuperEngine.runImpl(Engine.scala:535)
	at org.scalatest.wordspec.AnyWordSpecLike.run(AnyWordSpecLike.scala:1190)
	at org.scalatest.wordspec.AnyWordSpecLike.run$(AnyWordSpecLike.scala:1188)
	at akka.testkit.AkkaSpec.org$scalatest$BeforeAndAfterAll$$super$run(AkkaSpec.scala:59)
	at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
	at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
	at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
	at akka.testkit.AkkaSpec.run(AkkaSpec.scala:59)
	at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:318)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:513)
	at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by johanandren** [comment]
Should be fixed by #3647

**Comment by johanandren** [comment]
Didn't happen again after that fix so I'll close this
